src/components/structures/auth/Registration.tsx

```diff
@@ -141,6 +141,22 @@ export default class Registration extends React.Component<IProps, IState> {
         this.replaceClient(this.props.serverConfig);
     }
 
+    componentDidUpdate() {
+        if (this.state.doingUIAuth) {
+            //triggers a confirmation dialog for data loss before page unloads/refreshes
+            window.addEventListener("beforeunload", this.unloadCallback);
+        }
+    }
```

Comment:
this may cause multiple listeners to be bound, given the component may have multiple stages and thus updates, you should probably compare the state with prevState

Reply:
Thanks, done